### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by sind14**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,25 +1,79 @@
...
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: SlideLimitationValidator, name: str) -> None:
+        self.name = name
+        self.protected_name = f"_{name}"
+
+    def __get__(self,
+                instance: SlideLimitationValidator,
+                owner: SlideLimitationValidator) -> int | float:
```
Comment (olena-hul) [inline app/main.py]: owner is not SlideLimitationValidator

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 2]
```diff
@@ -1,25 +1,87 @@
...
+            )
+        except ValueError:
+            return False
+        return True
+
+
+visitor = Visitor("Arsen", 30, 100, 180)
+slide = Slide("Check Adult", AdultSlideLimitationValidator)
+result = slide.can_access(visitor)
+print(result)
```
Comment (olena-hul) [inline app/main.py]: Remove this debugging code

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,25 +1,81 @@
...
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self,
+                     owner: type[SlideLimitationValidator],
```
Comment (Bohdan-Salamakha) [inline app/main.py]: "type" is incorrect typehint

File: app/main.py [hunk 4]
```diff
@@ -1,25 +1,81 @@
...
+                     name: str) -> None:
+        self.name = name
+        self.protected_name = f"_{name}"
+
+    def __get__(self,
+                instance: SlideLimitationValidator,
+                owner: type[SlideLimitationValidator]) -> int | float:
```
Comment (Bohdan-Salamakha) [inline app/main.py]: "type" is incorrect typehint

File: app/main.py [hunk 5]
```diff
@@ -1,25 +1,81 @@
...
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(self,
+                 name: str,
+                 limitation_class: type[SlideLimitationValidator]) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py]: "type" is incorrect typehint

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v2

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**